docs/security.rst

```diff
@@ -123,6 +123,15 @@ Depending on the chosen mode (see :ref:`borg_init`) different primitives are use
 - The actual encryption is currently always AES-256 in CTR mode. The
   counter is added in plaintext, since it is needed for decryption,
   and is also tracked locally on the client to avoid counter reuse.
+
+  .. rubric:: Future plans
+
+  Adding support for session keys is planned. This would do away with the
+  tricky business of tracking counter usage and avoiding counter reuse and
+  allows to use different encryption algorithms in the same repository,
+  eg. to make encryption feasible on weaker devices or diversify the
+  available algorithms.
+
```

Comment:
(from the commit msg)

> since they tie into multi-threading as well i think it's
ok to say this out loud that we plan it.

I think in a document that is (supposed to) go into some detail about security design and practices in the project it's okay and imho even expected to mention some perspectives where we _want_ to improve.